spec/haskell/src/SEL4/Object/Reply.lhs

```diff
@@ -95,6 +101,7 @@ This module specifies the behavior of reply objects.
 >         setSchedContext scPtr (sc { scReply = prevReplyPtrOpt })
 >         when (prevReplyPtrOpt /= Nothing) $ do
 >             prevReplyPtr <- return $ fromJust prevReplyPtrOpt
+>             assert (replyPtr /= prevReplyPtr) "replyPop: reply lists must be distinct"
```

Comment:
I should have caught this before this PR was merged, but this line of assert, which is also added twice in `replyRemoveTCB`, does not seem to be needed at all. Why are these lines added to start with?

As I have been saying many weeks now, I am working on `reply_remove_tcb_corres`. So I’ve wasted some time on this while rebasing. Generally speaking, we probably want to make sure that all the spec updates are absolutely necessary. We really can’t afford to this kind of things, now for MCS in particular, but anytime, actually.